### PR TITLE
[PUB-2167] Save tags based on image 

### PR DESCRIPTION
**Pull request by albennett**
## Description
- Save tags based on the specific image. This fixes issue where tags were getting passed to different images in cases where the original image was deleted. 
https://buffer.atlassian.net/browse/PUB-2167
- Fix prop-type errors
- Allow user tagging access for all enterprise users

## Context & Notes
When editing a post, we need to add a set timeout when setting tags because of a timing issue with the draft images array. 

## Checklist

-   [ ] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have tested different user stories. 
-   [ ] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/composer-user-tags/components/UserTags/index.jsx [hunk 1]
```diff
@@ -213,14 +213,14 @@ UserTags.propTypes = {
   userTags: PropTypes.arrayOf(
     PropTypes.shape({
       username: PropTypes.string,
-      x: PropTypes.number,
-      y: PropTypes.number,
+      x: PropTypes.string,
+      y: PropTypes.string,
```
Comment (hamstu) [inline packages/composer-user-tags/components/UserTags/index.jsx]: Crush those warnings! 🔨

File: packages/composer/composer/action-creators/ComposerActionCreators.js [hunk 2]
```diff
@@ -403,9 +403,9 @@ const ComposerActionCreators = {
     });
   },
 
-  updateDraftUserTags: (id, userTags) => {
+  updateDraftImageUserTags: (id, userTags) => {
```
Comment (hamstu) [inline packages/composer/composer/action-creators/ComposerActionCreators.js]: Nice naming refactor! 👌

File: packages/composer/composer/components/MediaAttachmentThumbnail.jsx [hunk 3]
```diff
@@ -40,6 +40,9 @@ const renderTagOption = tagCount => (
   </StyledTagWrapper>
 );
 
+const getTags = draft =>
+  draft && draft.images && draft.images[0] && draft.images[0].userTags;
```
Comment (hamstu) [inline packages/composer/composer/components/MediaAttachmentThumbnail.jsx]: I can't wait until we finally get [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) in JS! Would make lines like this so much more simple. 😍

Comment (albennett) [reply packages/composer/composer/components/MediaAttachmentThumbnail.jsx]: Yassss 🙌 Can't wait for that!

**Pull request merged**